add.go

```diff
@@ -77,10 +77,14 @@ func (l *Conn) Add(addRequest *AddRequest) error {
 	defer l.finishMessage(messageID)
 
 	l.Debug.Printf("%d: waiting for response", messageID)
-	packet = <-channel
+	packetResponse, ok := <-channel
+	if !ok {
+		return NewError(ErrorNetwork, errors.New("ldap: channel closed"))
+	}
+	packet, err = packetResponse.ReadPacket()
 	l.Debug.Printf("%d: got response %p", messageID, packet)
-	if packet == nil {
```

Comment:
and are we sure there are no valid responses that return a nil packet? trying to think through the rfc...